### ISSUE TITLE
Python 3 status

### ISSUE THREAD

**Issue by AndrewC-B** [intrig-unicamp/mininet-wifi#170]
Hi,

Just wondering what the status of this project's support for Python 3 is? I've noticed a few bits of Python 3 incompatible code in `/mininet` (just a few uses of `basestring` so far). This is relevant to me because I don't need Wi-Fi emulation in all of my experiments, and I'd like to keep that code simple and cross-compatible with the original Mininet.

I'm happy to submit a pull request with the handful of corrections I've made so far, if that would be useful. You can view my changes [here](https://github.com/AndrewC-B/mininet-wifi/commit/8ee9643a1100510f5fe8c1f212ec102eda3a79b3).

Thanks!

**Comment by ramonfontes** [comment]
Hi,

I'll be happy to receive the pull request. It will be very useful for us. :)

**Comment by AndrewC-B** [comment]
Great! I’ll set that up when I next have a chance. The changes so far are very minor, but I’ll send any others I make your way too. 

I also ran into a few (Python 3) problems with the install script. The first is that `install.sh` uses the `python` command, rather than `python3`, even if passed the `-P` flag. As the default Python interpreter on Ubuntu 16.04 is Python 2.7 this causes a few problems. In particular I got an error (down in `setup.py`) along the lines of “no module typing” (`typing` is in Python 3 by default, but not Python 2). I worked around this by installing `typing` with pip before installing Mininet-WiFi.

If I have time I’ll see if I can chase down and fix some of these issues.
